src/drivers/MongoDriver.ts

```
@@ -1162,6 +1162,30 @@ export class MongoDriver implements DataStore {
       return Promise.reject('Error suggesting objects' + e);
     }
   }
+
+  async findSingleFile(params: {
+    learningObjectId: string;
+    fileName: string;
+  }): Promise<object> {
+    try {
+      const fileMetaData = await this.db.collection('objects').aggregate([
+        {$match: { '_id': params.learningObjectId }},
+        {$project: {
+          'materials.files': {$filter: {
+            input: '$materials.files',
+            as: 'file',
+            cond: {$eq: ['$$file.name', params.fileName]},
+          }},
+        }},
+      ]).toArray();
+
+      // Should never return more than one object.
```

Should we add a check to see if more than one file is returned? That way we can throw an error and catch that in Sentry if something weird starts happening. Thoughts?

---

Since only one element is needed in the result set, the `aggregate` function should be replaced in favor of MongoDB's `findOne` function which will only return one result. `findOne` can be used along with projection and the`$elemMatch` operator to achieve the filtering done within the `aggregate` function.